aten/src/ATen/native/sparse/cuda/SparseCsrTensorMath.cu

```diff
@@ -67,6 +68,38 @@ void convert_indices_from_coo_to_csr_cuda(const Tensor& result, const Tensor& in
   C10_CUDA_KERNEL_LAUNCH_CHECK();
 }
 
+template <typename input_t, typename output_t>
+__global__ void convert_indices_from_csr_to_coo_cuda_kernel(output_t* data_out, const input_t* data_in, const int64_t nrows) {
+  int64_t tid = blockDim.x * blockIdx.x + threadIdx.x;
+
+  if (tid < nrows) {
+    for (int64_t i = data_in[tid]; i < data_in[tid + 1]; i++)
+      data_out[i] = static_cast<output_t>(tid);
+  }
+}
+
+template <typename input_t, typename output_t>
+void convert_indices_from_csr_to_coo_cuda(const Tensor& result, const Tensor& crow_indices, const Tensor& col_indices) {
+  int64_t nrows = crow_indices.numel() - 1;
+  if (nrows == 0) {
+    result.zero_();
+    return;
+  }
+
+  auto crow_indices_ = crow_indices.expect_contiguous();
+  const input_t* crow_indices_data_in = crow_indices_->data_ptr<input_t>();
+  TORCH_INTERNAL_ASSERT(result.is_contiguous());
+  output_t* data_out = result.data_ptr<output_t>();
+
+  // Run nrows threads...
+  int64_t THREADS = at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock;
+  int64_t BLOCKS = (nrows + THREADS) / THREADS;
+  at::cuda::CUDAStream stream = at::cuda::getCurrentCUDAStream();
+  result.copy_(*col_indices.expect_contiguous());
```

Comment:
The copy should be applied to the second row of `result` only.

Reply:
Same as above.